infoextract.js

```diff
@@ -124,51 +144,56 @@ module.exports = {
 		});
 	},
 
+	getManyPreviews(videoIds, qParamsV) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				const videoInfo = await this.getVideoInfoYoutube(videoIds);
+				resolve(videoIds.map(id => videoInfo[id])
+						   	.filter(info => info !== undefined) //remove deleted videos
+						   	.map(info => {
+									if (qParamsV && qParamsV === info.id) {
+										info.highlight = true;
+									}
+									return info;
+								}));
+			} catch (err) {
+				console.error("Failed to compile add preview: error getting video info:", err);
+				reject(err);
+			}
+		});
+	},
+
 	getAddPreview(input) {
-		let service = this.getService(input);
+		const service = this.getService(input);
 
 		if (service !== "youtube") {
 			console.error("Unsupported input for getAddPreview");
 			throw "Unsupported input for getAddPreview";
 		}
 
-		let urlParsed = url.parse(input);
-		let queryParams = querystring.parse(urlParsed.query);
+		const urlParsed = url.parse(input);
+		const queryParams = querystring.parse(urlParsed.query);
 		if (queryParams["list"]) {
 			// there is a playlist associated with this link
 			console.log("playlist found");
 			return new Promise((resolve, reject) => {
 				this.getPlaylistYoutube(queryParams["list"]).then(playlist => {
-					let videoIds = playlist.map(item => item.id);
+					const videoIds = playlist.map(item => item.id);
 					console.log(`Found ${playlist.length} videos in playlist`);
-
-					this.getVideoInfoYoutube(videoIds).then(infoResults => {
-						let addPreviewResults = [];
-						for (let i = 0; i < videoIds.length; i++) {
-							const videoInfo = infoResults[videoIds[i]];
-							if (!videoInfo) {
-								// Failed to get info for this video
-								// video has probably been deleted, skip it
-								continue;
-							}
-							let video = videoInfo;
-							if (queryParams["v"] && video.id === queryParams["v"]) {
-								video.highlight = true;
-							}
-							addPreviewResults.push(video);
-						}
-						resolve(addPreviewResults);
-					}).catch(err => {
-						console.error("Failed to compile add preview: error getting video info:", err);
-						reject(err);
-					});
+					resolve(this.getManyPreviews(videoIds, queryParams["v"]));
 				}).catch(err => {
 					console.error("Failed to compile add preview: error getting playlist:", err);
 					reject(err);
 				});
 			});
-		}
-		else {
+		} else if (urlParsed.path.startsWith('/user') || urlParsed.path.startsWith('/channel')) {
+			console.log('channel found');
+			return new Promise((resolve, reject) => {
```

Comment:
There is no need to create another promise here, you can just return the promise from `getChannelInfoYoutube`

Reply:
There are several other instances of this used in your PR, please fix those as well.